src/yki/auth/code_auth.clj

```diff
@@ -18,11 +18,12 @@
     (if-let [login-link (login-link-db/get-login-link-by-code db (login-link/sha256-hash code))]
       (if (link-valid? login-link)
         (assoc
-         (found (:success_redirect login-link))
-         :session
-         {:identity {:external-user-id (:external_user_id login-link)}
-          :auth-method "EMAIL"
-          :yki-session-id (str (random-uuid))})
+          (found (:success_redirect login-link))
+          :session
+          {:identity       {:external-user-id (:external_user_id login-link)
+                            :email            (:email login-link)}
```

Comment:
Kun käyttäjä on alun perin kirjautunut Suomi.fi-tunnistautumisella, hänen participant-taulun riviin muodostuva external-user-id ei ole email, vaan OID tai hetu. Tällöin myöhemmin esim. maksulinkillä sovellukseen kirjautuessa sessioon asetettava external-user-id:n arvo ei ollut aiemman specin mukainen. Palautetaan tässä siis external-user-id:n lisäksi myös varsinainen email ja muutetaan spec vastaavasti tarkastelemaan email-kentän arvoa. external-user-id voidaan pitää mukana vastauksessa, jotta frontendin toteutusta ei tarvitse muuttaa synkassa backendin kanssa.